### PR TITLE
feat: persistent storage annotation

### PR DESCRIPTION
**Pull request by Falynsky**
-  Fix not working peristent storage with the getters
- Add `@LazyPersistentCached()` annotation
- Add `@PersistentCached()` annotation
- Add `@PersistentCached(initOnCall: )` parameter - initialize cache from external storage only after the first method call
- Mark `@Cached(persistentStorage: )` as @Deprecated
- Update documentation

### CODE REVIEW

**Review by AndriiChemer — APPROVED** [review]

File: packages/cached/lib/src/models/cached_method.dart [hunk 1]
```diff
@@ -4,11 +4,12 @@ import 'package:cached/src/models/cached_function.dart';
 import 'package:cached/src/models/cached_function_local_config.dart';
 import 'package:cached/src/models/param.dart';
 import 'package:cached/src/utils/asserts.dart';
+import 'package:cached_annotation/cached_annotation.dart';
 
 const _defaultSyncWriteValue = false;
 
 class CachedMethod extends CachedFunction {
-  CachedMethod._({
```
Comment (kamilraczkaiteo) [inline packages/cached/lib/src/models/cached_method.dart:11]: Does the constructor has to be public when it is being used only inside this class? Same with CachedGetter

File: packages/cached/lib/src/templates/cached_method_template.dart [hunk 2]
```diff
@@ -248,6 +279,16 @@ abstract class CachedMethodTemplate {
     ''';
   }
 
+  String _test() {
```
Comment (kamilraczkaiteo) [inline packages/cached/lib/src/templates/cached_method_template.dart:282]: Is the function needed? If yes, I think it should be named better

File: packages/cached/lib/src/models/cached_method.dart (hunk 1 above)
Comment (Falynsky) [reply packages/cached/lib/src/models/cached_method.dart:11]: It was needed but is no longer required, I forgot to change it back.

File: packages/cached/CHANGELOG.md [hunk 3]
```diff
@@ -1,8 +1,7 @@
 ## 1.7.0
 * Fix not working peristent storage with the getters
-* Add `@LazyPersistentCached()` annotation
-* Add `@PersistentCached()` annotation
-* Add `@PersistentCached(initOnCall: )` parameter - initialize cache from external storage only after the first method call
+* Add `@DirectPersistentCached()` annotation - avoid store
```
Comment (kamilraczkaiteo) [inline packages/cached/CHANGELOG.md:3]: Maybe slightly better explanation of `@DirectPersistentCached`?

File: packages/cached/README.md [hunk 4]
```diff
@@ -555,16 +571,27 @@ Cached library supports usage of any external storage (e.g. Shared Preferences,
   }
 ```
 
-`lazyPersistentStorage: true`
+### Init cache on method call
```
Comment (kamilraczkaiteo) [inline packages/cached/README.md:574]: I'd suggest to change it for:
```
## Lazy persistent storage
```

File: packages/cached/README.md [hunk 5]
```diff
@@ -555,16 +571,27 @@ Cached library supports usage of any external storage (e.g. Shared Preferences,
...
-  @Cached(lazyPersistentStorage: true)
-  Future<double> getLazyDouble() async {
+  @LazyPersistentCached()
+  Future<double> getDouble() async {
     return await _source.nextDouble() ;
   }
 ```
 
-When the `lazyPersistentStorage` parameter is used, it prevents the automatic loading of data from external storage into the cache managed by the caching library. For methods annotated with this parameter, the library's generator does not create a map for storing data fetched from the storage, nor does it initialize such a map before the method's invocation. Consequently, setting this parameter ensures that data is always fetched directly from externalStorage upon method call. If the data is not already present in externalStorage, it is retrieved and then stored there.
+## Direct persistent storage
```
Comment (kamilraczkaiteo) [inline packages/cached/README.md:585]: Both paragraphs `## Lazy persistent storage` and `## Direct persistent storage` seems to be mentioned too early. For sure there are mentioned before explanation what the standard `Persistent storage` does. I tend to find it confusing.
I'd suggest to firstly explain the Persistent Storage fully, as a feature, and then mention about other variants. You could move these two (`## Lazy persistent storage` and `## Direct persistent storage`) under the following text:
```
As you can see above, Cached doesn't provide any generic way of error or typing handling. It'll just use `PersistentStorageHolder.storage` to save and read cached data from storage in generated code. You have to take care of it yourself inside your code.  
```

**Review by kamilraczkaiteo — APPROVED** [review]

**Pull request merged**